pymbolic/primitives.py

```diff
@@ -1788,89 +1790,103 @@ def is_zero(value):
     return not is_nonzero(value)
 
 
-def wrap_in_cse(expr, prefix=None):
+def wrap_in_cse(expr: Expression,
+                prefix: str | None = None,
+                scope: str | None = None) -> Expression:
     if isinstance(expr, (Variable, Subscript)):
         return expr
 
+    if scope is None:
+        scope = cse_scope.EVALUATION
+
     if isinstance(expr, CommonSubexpression):
         if prefix is None:
             return expr
+
         if expr.prefix is None and type(expr) is CommonSubexpression:
-            return CommonSubexpression(expr.child, prefix)
+            return CommonSubexpression(expr.child, prefix, scope)
 
         # existing prefix wins
         return expr
 
     else:
-        return CommonSubexpression(expr, prefix)
+        return CommonSubexpression(expr, prefix, scope)
+
 
+def make_common_subexpression(expr: ExpressionT,
+                              prefix: str | None = None,
+                              scope: str | None = None) -> ExpressionT:
```

Comment:
Left `scope=None` here because setting it to `scope=cse_scope.EVALUATION` shows the actual string in the docs (i.e. `"pymbolic_eval"`), which are not shown in the docs for `cse_scope` itself.. very confusing :\